x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -797,9 +797,7 @@ export const mergeSearchResults = (searchResults: SignalSearchResponse[]) => {
       },
       aggregations: newAggregations,
       hits: {
-        total:
-          createTotalHitsFromSearchResult({ searchResult: prev }) +
-          createTotalHitsFromSearchResult({ searchResult: next }),
+        total: calculateTotal(prev, next),
```

Comment:
What was the motivation behind this refactor? It seems potentially unnecessary.

Reply:
because now createTotalHitsFromSearchResult returns `number | undefined` instead of just `number`, `calculateTotal` returns `-1` instead of `NaN`

Reply:
👍  It would be great to document this behavior with a unit test.